webrender/src/tiling.rs

```diff
@@ -105,6 +106,25 @@ impl AlphaBatchHelpers for PrimitiveStore {
         }
     }
 
+    fn get_blend_mode(&self, needs_blending: bool, metadata: &PrimitiveMetadata) -> BlendMode {
+        match metadata.prim_kind {
+            PrimitiveKind::TextRun => {
+                let text_run_cpu = &self.cpu_text_runs[metadata.cpu_prim_index.0];
+                match text_run_cpu.render_mode {
+                    FontRenderMode::Subpixel => BlendMode::Subpixel(text_run_cpu.color),
+                    FontRenderMode::Alpha | FontRenderMode::Mono => BlendMode::Alpha,
+                }
+            }
+            _ => {
+                if needs_blending {
+                    BlendMode::Alpha
+                } else {
+                    BlendMode::None
+                }
```

Comment:
I've left as is for now - will tidy up when I expand this function today for a different PR.